runtime/runtime-core/jvm/src/aws/smithy/kotlin/runtime/content/ByteStreamJVM.kt

```diff
@@ -33,6 +33,11 @@ fun File.asByteStream(start: Long = 0, endInclusive: Long = -1): ByteStream {
     return FileContent(this, start, endInclusive)
 }
 
+/**
+ * Create a [ByteStream] from a file with the given range
+ */
+fun File.asByteStream(range: LongRange) = asByteStream(range.first, range.last)
+
```

Comment:
**Comment**: Pending the outcome of the "do we need start/stop for `Path` discussion", I think we should include a `LongRange` overload for `Path` as well.